finagle-mysql/src/main/scala/com/twitter/finagle/mysql/Parameter.scala

```diff
@@ -0,0 +1,37 @@
+package com.twitter.finagle.exp.mysql
+
+import com.twitter.finagle.exp.mysql.transport.BufferWriter
+import language.implicitConversions
+
+trait Parameter {
+  type A
+  def value: A
+  def evidence: CanBeParameter[A]
```

Comment:
My opinion is that we probably want to keep the API surface as small as possible. And also, once you have an instance of `Parameter` you don't need an evidence that it can be parameter, since it's already it :)

As far as I know it's a common practice do not store evidences but wrap them into a closure. Although, it's probably just a nitpick. Feel free to ignore.